File: packages/redis/cache.ts
import { client } from "./client";
import type { WebsiteStatus } from "@repo/db/client";

export const setCurrentStatus = async (
    websiteId: string, 
    status : WebsiteStatus ) => {
    await client.set(
        `website:${websiteId}:status`,
        status,
        { EX : 60*10
        });
}

export const getPreviousStatusRedis = async (websiteId : string ) : Promise<WebsiteStatus | null>  => {
    const status = await client.get(`website:${websiteId}:status`);
    if(!status) return null;
    return status as WebsiteStatus;
}

File: apps/api-go/internal/redis/cache.go
package redis

import (
	"context"
	"fmt"
	"time"

	goredis "github.com/redis/go-redis/v9"

	"github.com/RitikaxG/runState/apps/api-go/internal/domain"
)

func (r *Redis) SetCurrentStatus(
	ctx context.Context,
	websiteId string,
	status domain.WebsiteStatus,
) error {
	key := fmt.Sprintf("website:%s:status", websiteId)

	return r.Client.Set(
		ctx,
		key,
		string(status),
		10*time.Minute,
	).Err()
}

func (r *Redis) getPreviousStatusRedis(
	ctx context.Context,
	websiteId string,
) (*domain.WebsiteStatus, error) {

	key := fmt.Sprintf("website:%s:status", websiteId)

	val, err := r.Client.Get(ctx, key).Result()
	if err != nil {
		if err.Error() == string(goredis.Nil) {
			return nil, nil
		}
		return nil, err
	}

	status := domain.WebsiteStatus(val)
	return &status, nil
}
